corehq/util/test_utils.py

```diff
@@ -195,3 +196,56 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self.logger.removeHandler(handler)
         for handler in self.original_handlers:
             self.logger.addHandler(handler)
+
+
+def generate_cases(argsets, cls=None):
```

Comment:
I recently did something similar to this, in a less generic way [here](https://github.com/dimagi/commcare-hq/blob/7d1af8a24edc67439aff28c9d6d5510f2b4208ef/corehq/ex-submodules/casexml/apps/phone/tests/test_extension_indexes.py#L37-L48). I like this way as it doesn't rely on an external file, however you don't have the ability to `./manage.py test TestThing.test_foo_with_bar`. 
Not sure if you think it is worth adding the other way as a reusable piece too?